# HG changeset patch
# Parent 5c83e2e03294a759f45858fabfee29048252cd29
Add cmake flag for building and using shared library

Controlled with
cmake -D USE_SHARED_POLARSSL_LIBRARY:BOOL=1 .

diff --git a/library/CMakeLists.txt b/library/CMakeLists.txt
--- a/library/CMakeLists.txt
+++ b/library/CMakeLists.txt
@@ -1,4 +1,4 @@
-add_library(polarssl STATIC
+set(src
      aes.c
      arc4.c
      base64.c
@@ -27,6 +27,17 @@
      xtea.c
 )
 
+if(NOT USE_SHARED_POLARSSL_LIBRARY)
+
+add_library(polarssl STATIC ${src})
+
+else()
+
+ADD_LIBRARY(polarssl SHARED ${src})
+SET_TARGET_PROPERTIES(polarssl PROPERTIES VERSION 1.0.0 SOVERSION 1)
+
+endif()
+
 INSTALL(TARGETS polarssl
         DESTINATION ${LIB_INSTALL_DIR}
         PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE)
